pkg/daemon/update.go

```diff
@@ -1455,6 +1455,16 @@ func createSSHKeyDir(authKeyDir string) error {
 		return exec.Command("runuser", "-u", constants.CoreUserName, "--", "mkdir", "-m", "0700", "-p", dir).Run()
 	}
 
+	// Check the existence of the /home/core/.ssh dir before creating a new one
+	// via runuser core by hand. Delete if any in case the dir is created at a wrong
+	// time and under the wrong user (root), and let the MCD recreate it.
+	// Serve as a workaround for https://issues.redhat.com/browse/OCPBUGS-11832
+	if _, err := os.Stat(constants.CoreUserSSHPath); err == nil {
```

Comment:
if err isn't nil here, maybe it would be better to return an error (as opposed to continuing)

It shouldn't really happen but from a code perspective I think it is better to catch errors